docs/samples/Microsoft.ML.Samples/Dynamic/GAM.cs

```diff
@@ -0,0 +1,91 @@
+
+            // Step 1: Read the data as an IDataView.
+            // First, we define the reader: specify the data columns and where to find them in the text file.
+            var reader = mlContext.Data.TextReader(new TextLoader.Arguments()
+                {
+                    Separator = "tab",
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoader.Column("MedianHomeValue", DataKind.R4, 0),
+                        new TextLoader.Column("CrimesPerCapita", DataKind.R4, 1),
+                        new TextLoader.Column("PercentResidental", DataKind.R4, 2),
+                        new TextLoader.Column("PercentNonRetail", DataKind.R4, 3),
+                        new TextLoader.Column("CharlesRiver", DataKind.R4, 4),
+                        new TextLoader.Column("NitricOxides", DataKind.R4, 5),
+                        new TextLoader.Column("RoomsPerDwelling", DataKind.R4, 6),
+                        new TextLoader.Column("PercentPre40s", DataKind.R4, 7),
+                        new TextLoader.Column("EmploymentDistance", DataKind.R4, 8),
+                        new TextLoader.Column("HighwayDistance", DataKind.R4, 9),
+                        new TextLoader.Column("TaxRate", DataKind.R4, 10),
+                        new TextLoader.Column("TeacherRatio", DataKind.R4, 11),
+                    }
+                });
+            
+            // Read the data
+            var data = reader.Read(dataFile);
+
+            // Step 2: Pipeline
+            // Concatenate the features to create a Feature vector.
+            // Then append a gam regressor, setting the "MedianHomeValue" column as the label of the dataset,
+            // the "Features" column produced by concatenation as the features column,
+            // and use a small number of bins to make it easy to visualize in the console window.
+            // For real appplications, it is recommended to start with the default number of bins.
+            var pipeline = mlContext.Transforms.Concatenate("Features", "CrimesPerCapita", "PercentResidental",
+                "PercentNonRetail", "CharlesRiver", "NitricOxides", "RoomsPerDwelling", "PercentPre40s",
+                "EmploymentDistance", "HighwayDistance", "TaxRate", "TeacherRatio")
+                    .Append(mlContext.Regression.Trainers.GeneralizedAdditiveModels(
+                        labelColumn: "MedianHomeValue", featureColumn: "Features", maxBins: 16));
+            var fitPipeline = pipeline.Fit(data);
+
+            // Extract the model from the pipeline
+            var gamModel = fitPipeline.LastTransformer.Model;
+
+            // Step 3: Investigate the properties of the model
+
+            // The intercept for the GAM models represent the average prediction for the training data
+            var intercept = gamModel.Intercept;
+            Console.WriteLine($"Average predicted cost: {intercept}");
```

Comment:
might want to put it in commented the output of this WriteLine.

Reply:
Resolved.